### PR TITLE
MINOR: Update test classes to use KafkaZkClient/AdminZkClient methods

### PR DESCRIPTION
**Pull request by omkreddy**

### CODE REVIEW

**Comment by omkreddy** [comment]
@junrao  pinging for review.

Will update remaining classes in next PR.

File: core/src/test/scala/unit/kafka/admin/AdminTest.scala [hunk 1]
```diff
@@ -110,6 +110,7 @@ class AdminTest extends ZooKeeperTestHarness with Logging with RackAwareTest {
 
   @Test
   def testTopicCreationInZK() {
+    zkClient.makeSurePersistentPathExists(BrokerIdsZNode.path)
```
Comment (ijuma) [inline core/src/test/scala/unit/kafka/admin/AdminTest.scala]: Why is this needed now and it wasn't before?

File: core/src/test/scala/unit/kafka/admin/DeleteTopicTest.scala [hunk 2]
```diff
@@ -67,21 +66,21 @@ class DeleteTopicTest extends ZooKeeperTestHarness {
...
+    TestUtils.verifyTopicDeletion(zkClient, topic, 1, servers)
   }
 
   @Test
   def testResumeDeleteTopicOnControllerFailover() {
     val topicPartition = new TopicPartition("test", 0)
     val topic = topicPartition.topic
     servers = createTestTopicAndCluster(topic)
-    val controllerId = zkUtils.getController()
+    val controllerId = zkClient.getControllerId.getOrElse(throw new KafkaException("Controller doesn't exist"))
```
Comment (ijuma) [inline core/src/test/scala/unit/kafka/admin/DeleteTopicTest.scala]: Nit: seems like `TestUtils.fail` might be more appropriate than `KafkaException`. A few other similar cases.

File: core/src/test/scala/unit/kafka/admin/DeleteTopicTest.scala [hunk 3]
```diff
@@ -176,7 +175,7 @@ class DeleteTopicTest extends ZooKeeperTestHarness {
 
   @Test
   def testAddPartitionDuringDeleteTopic() {
-    zkUtils.setupCommonPaths()
+    zkClient.createTopLevelPaths
```
Comment (ijuma) [inline core/src/test/scala/unit/kafka/admin/DeleteTopicTest.scala]: Nit: `()` is missing.

File: core/src/test/scala/unit/kafka/admin/AdminTest.scala [hunk 4]
```diff
@@ -310,7 +312,7 @@ class AdminTest extends ZooKeeperTestHarness with Logging with RackAwareTest {
     servers = TestUtils.createBrokerConfigs(2, zkConnect, false).map(b => TestUtils.createServer(KafkaConfig.fromProps(b)))
 
     // wait until reassignment completes
-    TestUtils.waitUntilTrue(() => !checkIfReassignPartitionPathExists(zkUtils),
```
Comment (ijuma) [inline core/src/test/scala/unit/kafka/admin/AdminTest.scala:313]: Can we delete `checkIfReassignPartitionPathExists`?

**Review by ijuma — COMMENTED** [review]
Thanks for the PR. Looks good overall, just a few minor comments. We can merge this once they are addressed.

File: core/src/test/scala/unit/kafka/admin/AdminTest.scala (hunk 1 above)
Comment (omkreddy) [reply core/src/test/scala/unit/kafka/admin/AdminTest.scala]: previously zkUtils.registerBrokerInZk -> zkCheckedEphemeral.create was creating the parent path recursively. Now zkClient.registerBrokerInZk expects parent path to present.

testcase for zkClient.registerBrokerInZk:
https://github.com/apache/kafka/blob/trunk/core/src/test/scala/unit/kafka/zk/KafkaZkClientTest.scala#L419

**Comment by omkreddy** [comment]
@ijuma  Thanks for the review. Updated the PR with review comments.

File: core/src/test/scala/unit/kafka/admin/AdminTest.scala (hunk 1 above)
Comment (ijuma) [reply core/src/test/scala/unit/kafka/admin/AdminTest.scala]: `TestUtils.createBrokersInZk` should create the parent path in that case IMO. I agree that `zkClient.registerBrokerInZk` should say as it is.

Comment (omkreddy) [reply core/src/test/scala/unit/kafka/admin/AdminTest.scala]: Agree. moved the  parent path creation to  parent path. Updated the PR.

**Comment by ijuma** [comment]
retest this please

**Review by ijuma — APPROVED** [review]
Thanks for the PR, LGTM.
